osf/management/commands/create_egap_registrations.py

```diff
@@ -0,0 +1,71 @@
+from datetime import datetime as dt
+
+from django.core.management.base import BaseCommand
+import django
+django.setup()
+
+from framework.auth.core import Auth
+from osf import exceptions
+from osf.models import DraftRegistration, OSFUser
+from osf.utils.permissions import ADMIN
+
+def register_silently(draft_registration, auth, sanction_type, external_registered_date, embargo_end_date):
+    try:
+        registration = draft_registration.register(auth, save=True)
+    except exceptions.NodeStateError as err:
+        raise exceptions.ValidationError(err)
+
+    registration.external_registered_date = external_registered_date
+
+    if sanction_type == 'Embargo':
+        try:
+            registration.embargo_registration(auth.user, embargo_end_date)
+        except exceptions.ValidationError as err:
+            raise exceptions.ValidationError(err.message)
+    else:
+        try:
+            registration.require_approval(auth.user)
+        except exceptions.NodeStateError as err:
+            raise exceptions.ValidationError(err)
+
+    registration.save()
+
+def main():
+    # Retrieve all EGAP Draft Registrations
+    egap_draft_registrations = DraftRegistration.objects.filter(registration_schema__name='EGAP Registration')
+
+    # Retrieve EGAP Admin (Greg?)
+    egap_author = OSFUser.objects.get(username='uday@cos.io')
```

Comment:
This username should probably be passed as a variable as with my script, don't want to expose Matt/Greg's username.

Reply:
Agreed - I changed it when integrating our management commands in the most recent commit.